### PR TITLE
Update project scaffolding and README

### PR DESCRIPTION
**Pull request by misscoded**
### Type of change (place an x in the [ ] that applies)

- [ ] New sample
- [ ] New feature
- [ ] Bug fix
- [x] Documentation + Refactor

### Summary

This PR updates the project structure offered to developers as a foundation upon which to build. In doing so, these changes bring alignment between [`bolt-*` starter templates](https://github.com/slack-samples/bolt-js-starter-template) and the Deno offering. 

The files and the samples contained within them are purposefully meaning agnostic while still offering enough functionality to serve as a solid jumping off point for development.

The README has also been completely rewritten to better suit the changes.

https://user-images.githubusercontent.com/7788242/187560210-0724b53a-952e-4892-b3b4-36432b30e9ce.mov

### Requirements (place an x in each [ ] that applies)

- [x] I’ve checked my submission against the Samples Checklist to ensure it complies with all standards
- [x] I have ensured the changes I am contributing align with existing patterns and have tested and linted my code
- [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct)

### CODE REVIEW

File: README.md
Comment (misscoded) [inline README.md:36]: A few reasons for the removal of a lot of this:

1. This template used to be blank, and so extensive instructions and explanations made sense. However, with the project changes, that information is no longer necessary: those structures are in place already.
2. As quickly as things change, having content embedded within the README is going to become untenable. We should be linking out to documentation as much as possible, which this document now does.

Comment (selfcontained) [reply README.md:36]: Totally makes sense. I think there are some gaps between what's current on the api docs site and what's being removed here (not that we shouldn't move forward to get this info shifted over to the api docs site). In the event that the existing Readme was useful in building out those api docs pages we can share a link to the prior version of the readme before we merged this removal in.

File: workflows/sample_workflow.ts
Comment (selfcontained) [inline workflows/sample_workflow.ts:36]: We could add `long: true` as a property on this field element, which I think results in a little nicer form experience for this use-case (even though it's quite simple).

image

**Review by selfcontained — APPROVED** [review]
Looks great, really like the new folder structure and baseline this provides to help educate or get going with.

**Review by hello-ashleyintech — COMMENTED** [review]
This looks great! Left some small questions about a few potential additions.

File: .gitignore
Comment (stevengill) [reply .gitignore:3]: Great suggestion to bring the GitHub workflow over! `CODEOWNERS` isn't a bad idea either, although would that be for the benefit of us as maintainers or would that be useful for devs who clone the template down? Looks like we just include it in the bolt-js starter template for folks cloning it down

File: manifest.ts
Comment (stevengill) [inline manifest.ts:10]: Nit: Accidental space on this line added

File: workflows/sample_workflow.ts
Comment (stevengill) [inline workflows/sample_workflow.ts:21]: Should leave some comments in this file. Like on line 20 here we could say something like "If your workflow requires collecting input from end users, we recommend using the openForm built-in function as a first step"

File: workflows/sample_workflow.ts
Comment (stevengill) [reply workflows/sample_workflow.ts:36]: ooo i didn't know that! Nice

**Review by stevengill — APPROVED** [review]
LGTM!

File: workflows/sample_workflow.ts
Comment (misscoded) [reply workflows/sample_workflow.ts:36]: I also had no idea this existed! I'll make the adjustment.

File: .gitignore
Comment (misscoded) [reply .gitignore:3]: I'll bring the GitHub workflow over! I hesitated on the CODEOWNERS file for the reason that @stevengill mentioned. Can we hold off on CODEOWNERS for the time being and discuss a bit more?

File: README.md
Comment (misscoded) [reply README.md:1]: Great suggestion! I'll include it.

File: workflows/sample_workflow.ts
Comment (misscoded) [reply workflows/sample_workflow.ts:21]: In complete agreement: we should add comments. I'll take a pass at it!

File: .gitignore
Comment (hello-ashleyintech) [reply .gitignore:3]: sounds good! @misscoded 👍 I had brought up `CODEOWNERS` more so for devs who clone the template down to build a sample app off of - definitely up for discussing this more though especially if that doesn't apply to the use case for this starter template!

File: README.md
Comment (misscoded) [reply README.md:1]: I went to go add it but it already exists in this project, so we should be good.

File: workflows/sample_workflow.ts
Comment (misscoded) [reply workflows/sample_workflow.ts:31]: Added comments to each component that aligns with the content we have in the README. I think it makes it much more usable and educational. Thanks for the suggestion!

File: functions/sample_function_test.ts
Comment (stevengill) [inline functions/sample_function_test.ts:7]: Test title is incorrect, but test itself looks good

Comment (misscoded) [reply functions/sample_function_test.ts:7]: Ah, good catch! 🤦🏼‍♀️

**Pull request merged**
